server/latest.go

```diff
@@ -0,0 +1,258 @@
+		if strings.Contains(*value.Key, fileSearchValue) && !strings.Contains(*value.Key, ".sig") {
+			switchValue := *value.Key
+			switch {
+			case strings.HasSuffix(switchValue, ".dmg"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-dmg")
+			case strings.HasSuffix(switchValue, ".exe"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-exe")
+			case strings.HasSuffix(switchValue, "amd64.deb"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-amd64-deb")
+			case strings.HasSuffix(switchValue, "i386.deb"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-i386-deb")
+			case strings.HasSuffix(switchValue, "x86_64.AppImage"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-x86_64-appimage")
+			case strings.HasSuffix(switchValue, "i386.AppImage"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-i386-appimage")
+			case strings.HasSuffix(switchValue, "x64.msi"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-x64-msi")
+			case strings.HasSuffix(switchValue, "x86.msi"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-x86-msi")
+			case strings.HasSuffix(switchValue, ver+"-linux-ia32.tar.gz"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-ia32-linux-tar")
+			case strings.HasSuffix(switchValue, ver+"-linux-x64.tar.gz"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-x64-linux-tar")
+			case strings.HasSuffix(switchValue, ver+"-linux-amd64.tar.gz"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-linux")
+			case strings.HasSuffix(switchValue, ver+"-windows-amd64.zip"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-windows")
+			case strings.HasSuffix(switchValue, ver+"-osx-amd64.tar.gz"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-osx")
+			}
+
+		}
+	}
+
+	return nil
+}
+
+func generateURLTextFile(cfg *MatterbuildConfig, params *s3.PutBucketWebsiteInput) (string, error) {
+	txtToReturn := ""
+	maxLen := len(params.WebsiteConfiguration.RoutingRules)
+	for len, value := range params.WebsiteConfiguration.RoutingRules {
+		txtToReturn += "https://" + cfg.S3BucketNameForLatestURLs + "/" + *value.Condition.KeyPrefixEquals
+		if len != maxLen-1 {
+			txtToReturn += "\n"
+		}
+	}
+
+	return txtToReturn, nil
+}
+
+func uploadIndexFile(awsSession client.ConfigProvider, cfg *MatterbuildConfig, txtFile string) error {
+	uploader := s3manager.NewUploader(awsSession)
+	result, err := uploader.Upload(&s3manager.UploadInput{
+		Bucket:      aws.String(cfg.S3BucketNameForLatestURLs),
+		Key:         aws.String("index.html"),
+		Body:        strings.NewReader(txtFile),
+		ContentType: aws.String("text/plain"),
+	})
+	if err != nil {
+		return errors.Wrapf(err, "failed to upload file, index.html")
+	}
+	LogInfo("File uploaded to, %s\n", result.Location)
+
+	return nil
+}
+
+// SetLatestURL updates the S3 website routing configuration
+func SetLatestURL(typeToRelease string, ver string, cfg *MatterbuildConfig) error {
```

Comment:
```suggestion
func setLatestURL(typeToRelease string, ver string, cfg *MatterbuildConfig) error {
```

Reply:
Since this is the entry function, for readability purposes, I would put it on top of the file.